packages/intent-filler/src/utils.ts

```diff
@@ -0,0 +1,23 @@
+import { ADDRESS_ZERO } from "hyperbridge-sdk"
+
+export async function fetchTokenUsdPriceOnchain(address: string, decimals: number): Promise<bigint> {
```

Comment:
I wonder if it makes sense to use something like tycho's universal router to fetch prices from actual onchain amms

Reply:
Tycho execution's router is currently deployed on Ethereum, Base and Unichain currently. 

I am using Coingecko because it has access to everychain's asset-prices.

We could use Uniswap's universal router for pricing. ![image](https://github.com/user-attachments/assets/3fcf50ca-fab8-49be-b4c2-3e512df8f582)